db/infinispan/src/main/java/org/commonjava/indy/infinispan/data/StoreDataManagerProvider.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (C) 2011-2020 Red Hat, Inc. (https://github.com/Commonjava/indy)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.infinispan.data;
+
+import org.commonjava.indy.conf.IndyConfiguration;
+import org.commonjava.indy.data.StandaloneStoreDataManager;
+import org.commonjava.indy.data.StoreDataManager;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+@ApplicationScoped
+public class StoreDataManagerProvider {
+
+    @Inject
+    StoreDataManager storeDataManager;
+
+    @Inject
+    @StandaloneStoreDataManager
+    InfinispanStoreDataManager infinispanStoreDataManager;
```

Comment:
If using the annotation inject, we either need to change the code to sth like '@cassandradatamanager' in future (and abandon this ispn annotation), or we have to inject both and pick one in the init() method (as below code shows). Either is not cool. IMO, a Producer method would be a better approach where you can instantiate proper manager according to the config.